### PR TITLE
create enrollment eligibility report with error handling

### PR DESCRIPTION
**Pull request by yosefserkez**
# PR Checklist

Please check if your PR fulfills the following requirements
- [ ] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (requires Feature flag)
- [x] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: 
TT6-186104308
# A brief description of the changes

Current behavior:
The script for the report is not in the code base, and it failed on errors.
New behavior:
The script is in the code base and it reports errors.
# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.
The original ticket is https://www.pivotaltracker.com/n/projects/2502930/stories/181021770/comments/229288499

### CODE REVIEW

File: lib/tasks/enrollment_eligibility_report.rake [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                    member_person.rating_address&.state,
+                    active_family_member_ids.include?(enrollment_member.applicant_id),
+                    applicant&.is_applying_coverage,
+                    applicant&.citizen_status,
+                    applicant&.is_incarcerated,
+                    applicant&.home_address&.state]
+          end
+        rescue => e
+          puts "Error: #{e.message}"
+          csv << [enrollment.hbx_id, enrollment.aasm_state, field_names[2..-2].map { |field| nil }, e.message]
```
Comment (mdkaraman) [inline lib/tasks/enrollment_eligibility_report.rake:45]: This line will create an entry in the third column that holds an array of nil values.  The error message will be in the fourth column.  I think the intention is to have the error message in the last column of the report, if present.  The arrays need to be concatenated and appended to the csv as a single array.

File: lib/tasks/enrollment_eligibility_report.rake [hunk 2]
```diff
@@ -0,0 +1,52 @@
+desc "Run a report for enrollment eligibility"
+task :enrollment_eligibility_report => :environment do
```
Comment (mdkaraman) [inline lib/tasks/enrollment_eligibility_report.rake:2]: Might be helpful to specify that this is used for enrollment renewal eligibility, e.g., `enrollment_renewal_eligibility_report.rake`

**Review by mdkaraman — CHANGES_REQUESTED** [review]

File: lib/tasks/enrollment_eligibility_report.rake (hunk 1 above)
Comment (4till2) [reply lib/tasks/enrollment_eligibility_report.rake:45]: ah yes we need the `.flatten`

File: lib/tasks/enrollment_renewal_eligibility_report.rake [hunk 3]
```diff
@@ -0,0 +1,52 @@
+desc "Run a report for enrollment eligibility"
+task :enrollment_renewal_eligibility_report => :environment do
+  file_name = "#{Rails.root}/enrollment_eligibility_report_#{TimeKeeper.date_of_record.strftime('%Y_%m_%d')}.csv"
+  field_names = %w[EnrollmentHbxId EnrollmentAASMState PrimaryHbxID MemberHbxID MemberFirstName MemberLastName MemberDOB member-is_applying_coverage member-citizen_status member-is_incarcerated member-state member_present_in_manage_family? applicant-is_applying_coverage applicant-citizen_status applicant-is_incarcerated applicant-state error_message]
+  CSV.open(file_name, 'w', force_quotes: true) do |csv|
+    csv << field_names
+    active_enrollments = HbxEnrollment.where(:aasm_state.nin => %w[shopping coverage_canceled coverage_terminated coverage_expired])
```
Comment (mdkaraman) [inline lib/tasks/enrollment_renewal_eligibility_report.rake:7]: Confirmed that the intended purpose of this script is to report specifically on IVL enrollments (including Coverall enrollments).  We can update this query to limit the enrollments like so:
```Ruby
active_enrollments = HbxEnrollment.where(:kind.in => %w[individual coverall], :aasm_state.nin => %w[shopping coverage_canceled coverage_terminated coverage_expired])
```

**Review by mdkaraman — APPROVED** [review]
Looks good!  The report is now client agnostic and targeted to IVL/Coverall enrollments.

**Review by RyanEddyIC — APPROVED** [review]

**Pull request merged**